src/DependencyInjection/CsaGuzzleExtension.php

```diff
@@ -123,4 +119,16 @@ private function processClientsConfiguration(array $config, ContainerBuilder $co
             }
         }
     }
+
+    private function findSubscriberIds(array $explicitlyConfiguredIds, ContainerBuilder $container)
+    {
+        $allNames = array_values(array_map(
+            function ($tags) { return $tags[0]['alias']; },
+            $container->findTaggedServiceIds(SubscriberPass::SUBSCRIBER_TAG)
+        ));
+
+        return array_filter($allNames, function ($name) use ($explicitlyConfiguredIds) {
+            return !isset($explicitlyConfiguredIds[$name]) || $explicitlyConfiguredIds[$name];
+        });
+    }
```

Comment:
By the way, I don't think we need to spend a lot of time on handling subscribers. Guzzle 6.0 now uses PSR-7, and middlewares instead. So let's keep it as is.

Thanks for your great job, @kgilden! :smile: